core/src/main/scala/org/apache/spark/status/LiveEntity.scala

```diff
@@ -641,8 +654,8 @@ private object LiveEntityHelpers {
         new v1.AccumulableInfo(
           acc.id,
           acc.name.map(weakIntern).orNull,
-          acc.update.map(_.toString()),
-          acc.value.map(_.toString()).orNull)
+          acc.update.map(accuValuetoString),
```

Comment:
Just found that in UI, the tasks of Stage tab shows `update` too. So we also need to truncate update value.

Reply:
Shown in the pr description.